packages/utils/src/index.js

```diff
@@ -153,3 +154,34 @@ export const createLazyDeferred = () => {
     },
   };
 };
+
+/**
+ * Given an array of functions which may throw a Promise when executed, we want
+ * to ensure all functions are executed, reducing any thrown Promises to a
+ * single Promise, which is itself rethrown.
+ * If no Promises are thrown, this is the equivalent of a .map
+ */
+export const captureSuspensePromises = executors => {
```

Comment:
See [the tests below](https://github.com/keystonejs/keystone-5/pull/1078/files#diff-6541306622514917f0cfbacaea6ae9a1R264)